onadata/apps/api/viewsets/data_viewset.py

```diff
@@ -393,6 +393,17 @@ def destroy(self, request, *args, **kwargs):
             if request.user.has_perm(CAN_DELETE_SUBMISSION, self.object.xform):
                 instance_id = self.object.pk
                 delete_instance(self.object, request.user)
+
+                # update xform if no instance has geoms
+                if (
+                    self.object.xform.instances.filter(
+                        deleted_at__isnull=True, geom=None
+                    ).count()
+                    < 1
+                ):
+                    self.object.xform.instances_with_geopoints = False
```

Comment:
Can we have this implemented on the model level instead of the viewset ? _Guessing this field is updated/set on the model level ?_

Reply:
@DavisRayM Moved the logic to the instance model post save signal instead to handle the update after a submission is soft deleted. Note that with soft deletion the post_save signal is triggered instead of post_delete